account_invoice_report_by_partner/models/account.py

```diff
@@ -0,0 +1,33 @@
+# encoding: utf-8
+##############################################################################
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as published
+#    by the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, api
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    @api.multi
+    def invoice_print(self):
+        self.ensure_one()

```

Comment:
Another thing I'm seeing now: you're restricting here the printing to only one invoice each time. You should at least warn about it in the README Known issues / Roadmap section. A direct improve is to allow to print when the invoice_report_id is the same (`if len(self.mapped('account_invoice_id')) == 1`)